enhancements/network/on-prem-configurable-lb-dns-stack.md

```diff
@@ -0,0 +1,284 @@
+It should support also the expected behaviors for "not in use" SLOs similar to the [CBO operator](https://github.com/openshift/enhancements/blob/master/enhancements/baremetal/an-slo-for-baremetal.md#standard-slo-behaviors).
+
+### cluster-hosted-net-services-operator details
+
+The cluster-hosted-net-services-operator should:
+
+- Be a new `openshift/cluster-hosted-net-services-operator` project.
+- Publish an image called `cluster-hosted-net-services-operator`.
+- Add a new `ClusterOperator` with an additional
+  `Disabled` status for non on-prem platforms.
+- Implement a controller reconciling a singleton instance representing the desired cluster hosted LB/DNS/VIP configuration.
+- Do nothing except set `Disabled=true`, `Available=true`, and
+  `Progressing=False` when the `Infrastructure` resource is a platform
+  type other than openstack, baremetal, ovirt or vsphere.
+- Update `ClusterOperator` DEGRADED field in accordance with the following healthchecks (in case self-hosted stack is enabled):
+  - single node holds the VIP
+  - kube-api is available via the VIP
+
+#### CRD definition
+
+Below is a possible option for a CRD instance of this operator:
+
+```yaml
+apiVersion: clusterHostedNetServices.operator.io/v1alpha1
+kind: HostedNetServices
+metadata:
+  name: cluster
+spec:
+  dns: 
+    nodesResolution: Enabled
+    appsResolution: Enabled
+    apiintIpAddress: 192.168.111.5 
+  ingressHa: Enabled
+status:
+  apiServerInternalIpOwner: master-0
+  ingressIpOwner: worker-1
+```
+
+With this sample manifest the cluster-hosted-net-services-operator should:
+
+- Create DNS hosted service that will:
+
+  - resolve node names and .apps wildcard record
+  - resolve api-int to 192.168.111.5.
+  - Run the self-hosted Loadbalancer for api only if apiintIpAddress is equal to API-VIP value provides in install-config file.
+  - Provide high availability for default ingress.
+
+From the status section we can see that master-0 node currently owns api-int VIP and worker-1 node owns ingress VIP. Additional spec and status fields may be introduced based on requirements.
+
+#### Managing  the cluster hosted networking services
+
+All the LB, DNS, and VIP components will be enabled by default in case of fresh deployment as today, while post-deployment the admin can control the services by editing the singleton CR.
+
+##### API load balancer migration
+
+Note: API LB migration is a significant infrastructure change that may cause an API outage for some time. There is no requirement for the process to take place without service disruption.
+
+-- self-hosted to an external Load balancer
+
+In order to migrate the self-hosted LB to an external Load balancer the admin should:
+
+- Provide new IP address (!= API-VIP from install-config) pointing to external Load balancer front end.
```

Comment:
is there no way to use a DNS name only?

Reply:
I don't think it's possible since there are kubeconfig files (e.g: /var/lib/kubelet/kubeconfig) pointing to https://api-int.ostest.test.metalkube.org:6443 server